contracts/pools/steth/StableSwapSTETH.vy

```diff
@@ -0,0 +1,832 @@
+    dy = dy - dy_fee
+    assert dy >= min_dy, "Exchange resulted in fewer coins than expected"
+
+    admin_fee: uint256 = self.admin_fee
+    if admin_fee != 0:
+        dy_admin_fee: uint256 = dy_fee * admin_fee / FEE_DENOMINATOR
+        if dy_admin_fee != 0:
+            self.admin_balances[j] += dy_admin_fee
+
+    coin: address = self.coins[1]
+    if i == 0:
+        assert msg.value == dx
+        assert ERC20(coin).transfer(msg.sender, dy)
+    else:
+        assert msg.value == 0
+        assert ERC20(coin).transferFrom(msg.sender, self, dx)
+        raw_call(msg.sender, b"", value=dy)
+
+    log TokenExchange(msg.sender, i, dx, j, dy)
+
+    return dy
+
+
+@external
+@nonreentrant('lock')
+def remove_liquidity(
+    _amount: uint256,
+    _min_amounts: uint256[N_COINS],
+) -> uint256[N_COINS]:
+    """
+    @notice Withdraw coins from the pool
+    @dev Withdrawal amounts are based on current deposit ratios
+    @param _amount Quantity of LP tokens to burn in the withdrawal
+    @param _min_amounts Minimum amounts of underlying coins to receive
+    @return List of amounts of coins that were withdrawn
+    """
+    amounts: uint256[N_COINS] = self._balances()
+    lp_token: address = self.lp_token
+    total_supply: uint256 = ERC20(lp_token).totalSupply()
+    CurveToken(lp_token).burnFrom(msg.sender, _amount)  # dev: insufficient funds
+
+    for i in range(N_COINS):
+        value: uint256 = amounts[i] * _amount / total_supply
+        assert value >= _min_amounts[i], "Withdrawal resulted in fewer coins than expected"
+
+        amounts[i] = value
+        if i == 0:
+            raw_call(msg.sender, b"", value=value)
+        else:
+            assert ERC20(self.coins[1]).transfer(msg.sender, value)
+
+    log RemoveLiquidity(msg.sender, amounts, empty(uint256[N_COINS]), total_supply - _amount)
+
+    return amounts
+
+
+@external
+@nonreentrant('lock')
+def remove_liquidity_imbalance(
+    _amounts: uint256[N_COINS],
+    _max_burn_amount: uint256
+) -> uint256:
+    """
+    @notice Withdraw coins from the pool in an imbalanced amount
+    @param _amounts List of amounts of underlying coins to withdraw
+    @param _max_burn_amount Maximum amount of LP token to burn in the withdrawal
+    @return Actual amount of the LP token burned in the withdrawal
+    """
+    assert not self.is_killed  # dev: is killed
+
+    amp: uint256 = self._A()
+    old_balances: uint256[N_COINS] = self._balances()
+    D0: uint256 = self.get_D(old_balances, amp)
+    new_balances: uint256[N_COINS] = old_balances
+    for i in range(N_COINS):
+        new_balances[i] -= _amounts[i]
+    D1: uint256 = self.get_D(new_balances, amp)
+
+    fees: uint256[N_COINS] = empty(uint256[N_COINS])
+    fee: uint256 = self.fee * N_COINS / (4 * (N_COINS - 1))
+    admin_fee: uint256 = self.admin_fee
+    for i in range(N_COINS):
+        ideal_balance: uint256 = D1 * old_balances[i] / D0
+        new_balance: uint256 = new_balances[i]
+        difference: uint256 = 0
```

Comment:
Oh interesting. Declaring `difference` inside the loop ended up being more optimal?